### PR TITLE
Restrict ldap group update to prevent overwriting existing data.

### PR DESCRIPTION
**Pull request by henning-gerhardt**
Prevent overwriting of existing data which was introduced by #1870 .

### CODE REVIEW

File: Kitodo-DataManagement/src/main/resources/db/migration/V2_54__update_LDAP_Group_table.sql [hunk 1]
```diff
@@ -29,4 +29,4 @@ sambaPasswordMustChange = '2147483647',
 sambaPasswordHistory = '00000000000000000000000000000000000000',
 sambaLogonHours = 'FFFFFFFFFFFFFFFFFFFF',
 sambaKickoffTime = '0'
-WHERE id = 2 ;
+WHERE id = 2 AND homeDirectory IS NULL AND gidNumber IS NULL AND sambaSID IS NULL;
```
Comment (Kathrin-Huber) [inline Kitodo-DataManagement/src/main/resources/db/migration/V2_54__update_LDAP_Group_table.sql]: does the id needs to be 2?

Comment (henning-gerhardt) [reply Kitodo-DataManagement/src/main/resources/db/migration/V2_54__update_LDAP_Group_table.sql]: I did not add id with value 2 as this was in the previous and merged pull request. My goal is only not to overwrite existing data.

**Comment by henning-gerhardt** [comment]
I noticed yet that there is maybe even an wrong data inserted for homeDirectory (see my comment in #1870 ). Should this be fixed in this pull request or in a new migration file?

File: Kitodo-DataManagement/src/main/resources/db/migration/V2_54__update_LDAP_Group_table.sql (hunk 1 above)
Comment (Beacze) [reply Kitodo-DataManagement/src/main/resources/db/migration/V2_54__update_LDAP_Group_table.sql]: I think default one has name test, so maybe we could use name.

**Comment by Beacze** [comment]
This migration should be updated. Adding new one is not necessary.

File: Kitodo-DataManagement/src/main/resources/db/migration/V2_54__update_LDAP_Group_table.sql (hunk 1 above)
Comment (Kathrin-Huber) [reply Kitodo-DataManagement/src/main/resources/db/migration/V2_54__update_LDAP_Group_table.sql]: I think searching by name is the better idea here.

Comment (henning-gerhardt) [reply Kitodo-DataManagement/src/main/resources/db/migration/V2_54__update_LDAP_Group_table.sql]: Ok. I will change it.

**Comment by henning-gerhardt** [comment]
Changing WHERE condition on migration from id to title did not work:

```
[ERROR] Message    : You are using safe update mode and you tried to update a table without a WHERE that uses a KEY column
[ERROR] Location   : Kitodo-DataManagement/src/main/resources/db/migration/V2_54__update_LDAP_Group_table.sql (/home/travis/build/henning-gerhardt/kitodo-production/Kitodo-DataManagement/src/main/resources/db/migration/V2_54__update_LDAP_Group_table.sql)
```
So I must stay with `id = 2`

File: Kitodo-DataManagement/src/main/resources/db/migration/V2_54__update_LDAP_Group_table.sql (hunk 1 above)
Comment (henning-gerhardt) [reply Kitodo-DataManagement/src/main/resources/db/migration/V2_54__update_LDAP_Group_table.sql]: Did not work as flyway migration complain about missing KEY column in WHERE condition.

**Comment by Beacze** [comment]
Switch off safe update: 
SET SQL_SAFE_UPDATES = 0;

**Pull request merged**